client/lib/jetpack/use-threats.ts

```diff
@@ -12,7 +12,21 @@ import getSiteScanUpdatingThreats from 'calypso/state/selectors/get-site-scan-up
 
 export const useThreats = ( siteId: number ) => {
 	const [ selectedThreat, setSelectedThreat ] = useState< Threat >();
-	const updatingThreats = useSelector( ( state ) => getSiteScanUpdatingThreats( state, siteId ) );
+	const updatingThreats = useSelector(
+		( state ) => {
+			return getSiteScanUpdatingThreats( state, siteId );
+		},
+		( a, b ) => {
+			// Compare arrays by reference first, then by contents
+			if ( a === b ) {
+				return true;
+			}
+			if ( ! a || ! b || a.length !== b.length ) {
+				return false;
+			}
+			return a.every( ( val, idx ) => val === b[ idx ] );
+		}
+	);
```

Comment:
should fix this warning:
> index.ts:56 Selector unknown returned a different result when called with the same parameters. This can lead to unnecessary rerenders.
Selectors that return a new reference (such as an object or an array) should be memoized: https://redux.js.org/usage/deriving-data-selectors#optimizing-selectors-with-memoization 

(together with the introduced `createSelector()` calls elsewhere in this PR)